### PR TITLE
Feature:  add subdomains limit

### PR DESCRIPTION
**Pull request by jotaesteves**
- adds subdomains nested store from domainStore
- limits the creation of subdomains by level trought `.env`variable  `CONTEXTURE_MAX_SUBDOMAINS_NESTING_LEVEL` currently set to 1.
- hides create subdomain action

image
 
image

### CODE REVIEW

File: frontend-vue/src/pages/domain-details/DomainDetails.vue [hunk 1]
```diff
@@ -261,6 +265,16 @@ function onEditCloseClick() {
   editMode.value = false;
 }
 
+watch(
```
Comment (thomas-burko) [inline frontend-vue/src/pages/domain-details/DomainDetails.vue:268]: Personally I strive to avoid using watch() too often as it makes it hard to follow the data flow. Can we use a computed property instead?

const canCreateSubdomain = computed(() => { /* recalculate the relative level to parent /*});

File: frontend-vue/src/stores/subDomains.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
+/* export a nested subdomains store from the root domains store */
+import { defineStore } from "pinia";
+import { computed, ref } from "vue";
+import { Domain, DomainId } from "~/types/domain";
+import { useDomainsStore } from "~/stores/domains";
+
+export const useSubdomainsStore = defineStore("subdomains", () => {
```
Comment (thomas-burko) [inline frontend-vue/src/stores/subDomains.ts:7]: Do we need a seperate store for subdomains or can we reuse the existing one? The naming is confusing because we don't only handle subdomains but also domains (parent).

The implementation seems to have a bug. If I do the following:
All Domains List -> Domain A -> Navigate Back (via Breadcrumbs or back button) -> Domain A = I cannot create a new subdomain.

**Two possible solutions come to mind:**
**Lookup**
Depending on how expensive it is to calculate a level for a domain it may make sens to create a initial lookup map similar to https://vscode.dev/github/trustbit/Contexture/blob/feature/beschr%C3%A4nkung-der-verschachtelungstiefe-von-subdomains/frontend-vue/src/stores/domains.ts#L112. Because we load all domains anyway for now.

So you would have a <DomainId, Level> mapping and only need to lookup based on this and don't need to recalculate it everytime you open a page (And don't need the init handling). You could also reuse the existing computed values like `domainByDomainId` and `subdomainsByDomainId` to quickly access a domain by its id instead of iterating over it.

**Disregard the store**
Reuse the existing domain store and do the domain level calculation on a component/service level. You would recalculate the level everytime you navigate to the page which may make sense if we go away from loading all domains at once.

**Review by thomas-burko — CHANGES_REQUESTED** [review]
🚀 Nice UX change. Please see my comments below.

File: frontend-vue/src/stores/subDomains.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    domainLevel.value++;
+
+    return {
+      isSubdomain,
+      domainLevel,
+    };
+  }
+
+  const isRootSubdomain = computed(() => {
```
[the remark this answers is not in the document]
Comment (jotaesteves) [reply frontend-vue/src/stores/subDomains.ts:49]: Yes, it's removed now.

File: frontend-vue/src/stores/subDomains.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    const parentDomainId = currentDomain.value?.parentDomainId || false;
+    return parentDomainId && domainLevel.value === 2;
+  });
+
+  const isCreateSubdomainDisabled = computed(() => {
```
[the remark this answers is not in the document]
Comment (jotaesteves) [reply frontend-vue/src/stores/subDomains.ts:54]: no problem, changed it to improve readability

File: frontend-vue/src/pages/domain-details/DomainDetails.vue (hunk 1 above)
Comment (jotaesteves) [reply frontend-vue/src/pages/domain-details/DomainDetails.vue:268]: I agree with you, in this case to ensure the render/logic is updated first and consistently the `watch` property here works as expected.
Alternatively we could implement a Middleware for route changes altough this could lead to overengineering this logic.

File: frontend-vue/src/stores/subDomains.ts (hunk 2 above)
Comment (jotaesteves) [reply frontend-vue/src/stores/subDomains.ts]: Good point and good catch on the side effect of the first implementation, 
I've removed the `subDomains` store and changes the code accordingly to utilize 

> <DomainId, Level> mapping 

The side-effect is fixed and all code is in `domains.ts`store

**Review by thomas-burko — APPROVED** [review]
